goosebit/settings/const.py

```diff
@@ -1,8 +1,11 @@
 from pathlib import Path
 
+import platformdirs
 from argon2 import PasswordHasher
 
 BASE_DIR = Path(__file__).resolve().parent.parent.parent
+CONFIG_DIR = platformdirs.site_config_path("gooseBit")
```

Comment:
Not a fan :disappointed:
```
>>> platformdirs.site_config_path("gooseBit")
PosixPath('/etc/xdg/xdg-ubuntu/gooseBit')
```

Is it not possible to just get `/etc/goosebit.yaml` or maybe `/etc/goosebit/settings.yaml`?

Reply:
Not sure.  I don't know why platformdirs wants these weird paths, I'll look into what other options we have there.

Reply:
I think the only other option for this might be `user_config_path`, I checked every other platformdirs call, and none of them look any better.  This resolves to `~/.config/gooseBit` when called with `user_config_path("gooseBit)` for me.

Reply:
On Linux/Unix, searching these locations in the following order would be fine:
  1. `settings.yaml` in `$PWD` (or relative to the code) for development and people running it from e.g. `/app` in a container
  1. `~/.config/goosebit/settings.yaml` for a root-less installation
  1. `/etc/goosebit/settings.yaml` for a system-wide installation

For Windows, the first one should already be enough, so we could also just hardcode those.

Reply:
> For Windows, the first one should already be enough, so we could also just hardcode those.

I think we found out with the whole crowdstrike incident how many people are using Windows for things they shouldn't really be, so I'm not certain how much we want to support Windows.  This is essentially a breaking change for using it with `pip` in windows, which may be annoying if trying to test.

Overall up to you, I like the directories you listed, but I am generally of the opinion that trying to support all major systems as well as possible is a good idea.

Reply:
What I tried to say is, if we support loading `settings.yaml` from the current working directory, a clean deployment can be achieved on Windows. No need to search `%AppData%` or whatever.

On second thought, why not try `os.getcwd()` and then fallback to the `GOOSEBIT_SETTINGS` environment variable. That keeps it simple and flexible at the same time.

Reply:
> What I tried to say is, if we support loading `settings.yaml` from the current working directory, a clean deployment can be achieved on Windows. No need to search `%AppData%` or whatever.
> 
> On second thought, why not try `os.getcwd()` and then fallback to the `GOOSEBIT_SETTINGS` environment variable. That keeps it simple and flexible at the same time.

Yeah this seems like the best solution.  The only problem is we are loading our environment variables with the same loader that loads the yaml file, but I think I can make another one, load it, then use it for the other one.

Regarding directory default, do you think going `os.cwd()`, then goosebit root, then `GOOSEBIT_SETTINGS` will work best?  Or do we just ignore the root and hope that development envs are running it properly?

Reply:
> Yeah this seems like the best solution. The only problem is we are loading our environment variables with the same loader that loads the yaml file, but I think I can make another one, load it, then use it for the other one.

Just do a `os.getenv()` for that one.

> Regarding directory default, do you think going `os.cwd()`, then goosebit root, then `GOOSEBIT_SETTINGS` will work best? Or do we just ignore the root and hope that development envs are running it properly?

I guess also checking the project root doesn’t hurt.

Reply:
Sounds good.  Going to bed for now, but will take another stab at this tomorrow.